File: python/orientacao.py
from compromisso import Compromisso

class Orientacao(Compromisso):
    def __init__(self, id, data, hora, duracao, prioridade, adiavel, nomeOrientado, nivel, assunto):
        super().__init__(id, 'O', data, hora, duracao, prioridade, nomeOrientado)
        self.adiavel = adiavel
        self.nivel = nivel
        self.assunto = assunto
        self.calcularPrioridadeFinal()
    
    def calcularPrioridadeFinal(self):
        multiplicador = 1
        if not self.adiavel:
            multiplicador += 1
        self.prioridadeFinal = self.prioridade * multiplicador
    
    def getDescricao(self):
        return f"Orientação de {self.descricao}"
    
    def isAdiavel(self):
        return self.adiavel
    
    def getNivel(self):
        return self.nivel
    
    def getAssunto(self):
        return self.assunto

File: C++/src/orientacao.cpp
#include "orientacao.h"

Orientacao::Orientacao(){}

Orientacao::Orientacao(int id, const Data& data, const Hora& hora, int duracao, int prioridade, bool adiavel, const string& descricao, const string& nivel, const string& assunto)
    : Compromisso(id, 'O', data, hora, duracao, prioridade, prioridade, descricao), adiavel(adiavel), nivel(nivel), assunto(assunto){}

string Orientacao::getDescricao() const{
    return "Orientação de " + descricao;
}

void Orientacao::calcularPrioridadeFinal(){
    int fator = 1;
    if(!adiavel) fator += 1;
    prioridade_final = prioridade * fator;
}

string Orientacao::getNivel() const{
    return nivel;
}

string Orientacao::getAssunto() const{
    return assunto;
}